contributors/design-proposals/controller_history.md

```diff
@@ -0,0 +1,447 @@
+
+### History Maintenance
+Controllers should be configured, either globally or on a per specification type 
+Object basis, to have a `RevisionHistoryLimit`. This field will indicate the 
+number of non-live revisions the controller should maintain in its history
+for each specification type Object. Every time a controller observes a 
+specification type Object it will do the following.
+
+1. The controller will 
+[reconstruct the Object's revision history](#history-reconstruction).
+    - Note that the process of reconstructing the Object's history filters any
+    AnyStates not owned by the Object.
+1. The controller will filter any AnyStates that represent a live version.
+1. If the number of remaining AnyStates is greater than the configured 
+`RevisionHistoryLimit`, the controller will delete them, in order with respect 
+to the value mapped to their `"ControllerGeneration"` keys, until the number 
+of remaining AnyStates is equal to the `RevisionHistoryLimit`.
+
+This ensures that the number of recorded, non-live revisions is less than or 
+equal to the configured `RevisionHistoryLimit`.
+
+### Version Tracking
+Controllers must track the version of the target Object state that corresponds 
+to their generated Objects. This information is necessary to determine which 
+versions are live, and to track which Objects need to be updated during a 
+target state update or rollback. We propose two methods that controllers may 
+use to track live versions and their association with generated Objects.
+
+1. The most straightforward method is labeling. In this method the generated 
+Objects are labeled with the `.Name` of the AnyState object that corresponds to 
+the version of the target Object state that was used to generate them. As we 
+have taken care to ensure the uniqueness of the `.Names` of the AnyStates, 
+this approach is reasonable.
+    - A revision is considered to be live while any generated Object labeled 
+     with its `.Name` is live.
+    - This method has the benefit of providing visibility, via the label, to 
+    users with respect to the historical providence of a generated Object.
+    - The primary drawback is the lack of support for using garbage collection 
+    to ensure that only non-live version snapshots are collected.
+1. Controllers may also use the `OwnerReferences` field of the AnyState to 
+record all Objects that are generated from target Object state version 
+represented by the AnyState as its owners.
+   - A revision is considered to be live while any generated Object that owns 
+   it is live.
+   - This method allows for the implementation of generic garbage collection.
+   - The primary drawback with this method is that the book keeping is complex, 
+   and deciding if a generated Object corresponds to a particular revision 
+   will require testing each Object for membership in the `OwnerReferences` 
+   of all AnyStates.
+   
+At the cost of the complexity of implementing both labeling and ownership, 
```

Comment:
I've been putting some more thought into this. I did not consider adoption initially because I was coming from the viewpoint that controller history is not adoptable. I still have not convinced myself that it makes sense to adopt the history of another object, but I haven't convinced myself that this is an aweful idea either.